File: csharp/booking-up-for-beauty/BookingUpForBeauty.cs
using System;

static class Appointment
{
    public static DateTime Schedule(string appointmentDateDescription) => DateTime.Parse(appointmentDateDescription);

    public static bool HasPassed(DateTime appointmentDate) => DateTime.Now > appointmentDate;
    
    public static bool IsAfternoonAppointment(DateTime appointmentDate) => 12 <= appointmentDate.Hour && appointmentDate.Hour < 18;

    public static string Description(DateTime appointmentDate) => $"You have an appointment on {appointmentDate.ToString()}.";

    public static DateTime AnniversaryDate() => new DateTime(DateTime.Today.Year, 9, 15);
}

File: fsharp/booking-up-for-beauty/BookingUpForBeauty.fs
module BookingUpForBeauty

// The following line is needed to use the DateTime type
open System

let schedule (appointmentDateDescription: string): DateTime = DateTime.Parse appointmentDateDescription

let hasPassed (appointmentDate: DateTime): bool = appointmentDate.CompareTo DateTime.Now < 0

let isAfternoonAppointment (appointmentDate: DateTime): bool = 12 <= appointmentDate.Hour && appointmentDate.Hour < 18

let description (appointmentDate: DateTime): string = $"""You have an appointment on {appointmentDate.ToString("M/d/yyyy h:mm:ss tt")}.""";

let anniversaryDate(): DateTime = failwith "Please implement the 'anniversaryDate' function"
